base/abstractarray.jl

```diff
@@ -2049,19 +2050,28 @@ map(f, ::AbstractSet) = error("map is not defined on sets")
 ## 2 argument
 function map!(f::F, dest::AbstractArray, A::AbstractArray, B::AbstractArray) where F
     for (i, j, k) in zip(eachindex(dest), eachindex(A), eachindex(B))
-        dest[i] = f(A[j], B[k])
+        @inbounds a, b = A[j], B[k]
+        val = f(a, b)
+        @inbounds dest[i] = val
     end
     return dest
 end
 
 ## N argument
 
 @inline ith_all(i, ::Tuple{}) = ()
-@inline ith_all(i, as) = (as[1][i], ith_all(i, tail(as))...)
+function ith_all(i, as)
```

Comment:
Are we sure it's ok to drop `@inline` here?

Reply:
Doesn't `@_propagate_inbounds_meta` imply inlining?

Reply:
https://github.com/JuliaLang/julia/blob/7772486a6942c9a9b52849b6ffe690a7da5861cd/base/essentials.jl#L99-L101